medusa/search/core.py

```diff
@@ -938,8 +947,8 @@ def combine_results(multi_results, single_results):
         # see how many of the eps that this result covers aren't covered by single results
         needed_eps = []
         not_needed_eps = []
-        for result in single_results:
-            if result in multi_result.episodes:
+        for result in return_single_results:
+            if result.episodes[0] in multi_result.episodes:
```

Comment:
Previously you where comparing are TorrentSearch object with a list of episode objects. Which was always `False` of course.